### ISSUE TITLE
Inconsistent secondary structure between -dssp and -ss

### ISSUE THREAD

**Issue by jbarnoud** [Tsjerk/Martinize#7]
**tl;dr** The secondary sequence written in the ITP file is not what the `-ss` option expect and it is confusing.

This was reported to me by Alex.

I ran martinize in two different ways that should give the exact same result, yet the secondary structure that martinize uses in its output differs between the two runs:

* First I ran martinize using the `-dssp` option:
```
martinize -f /home/jon/dev/Martinize/tests/data/inputs/1ubq/1ubq.pdb -o system-vaccum.top -x 1UBQ-CG.pdb -dssp dssp -p backbone -ff martini22
```
* Then I extracted the secondary structure sequence that the first run wrote in `Protein_A.itp` and ran martinize with the `-ss` option and that sequence:
```
martinize -f /home/jon/dev/Martinize/tests/data/inputs/1ubq/1ubq.pdb -o system-vaccum.top -x 1UBQ-CG.pdb -ss CEEEEEETTSCEEEEECCTTSC1111HHHH2222CCCCCCEEEEETTEECCTTSCTCCCTCCTTCEEEEEECCSCC -p backbone -ff martini22
```

While the output should be identical, the secondary structure actually differ:

```
# Extract of Protein_A.itp using -dssp
; Secondary Structure:
; CEEEEEETTSCEEEEECCTTSC1111HHHH2222CCCCCCEEEEETTEECCTTSCTCCCTCCTTCEEEEEECCSCC
# Extract using -ss
; Secondary Structure:
; CEEEEEETTSCEEEEECCTTSCCCCC3333CCCCCCCCCCEEEEETTEECCTTSCTCCCTCCTTCEEEEEECCSCC
```

Of course, because the SS sequence differs, the resulting topology differs as well.

Here is the segment of the SS sequence that differs (first dssp, then ss):
```
1111HHHH2222
CCCC3333CCCC
```

It appears that helix starts and ends (`1` and `2`, respectively) are replaced by coil (`C`) while the helix structure (`H`) is replaced by a short helix (`3`).

From reading the source code, I understand that what is written in the ITP file header is what is stored in `IO.Chain.sstypes`. When this same string is provided through the `-ss` option, it is stored in `IO.Chain.ss` and transformed by `secstruct..ssClassification` with the source being `self`. From what I understand, a secondary structure coming from the program "self" should be left untouched when translated to secondary types here:

https://github.com/Tsjerk/Martinize/blob/44445a7dd60b1ad66e6ba6d924a68e041d873959/martinize/secstruc.py#L148

However, the translation only knows about "F", "H", "E", "T", "S", and "C". Everything else is translated to "C":

https://github.com/Tsjerk/Martinize/blob/44445a7dd60b1ad66e6ba6d924a68e041d873959/martinize/secstruc.py#L64

The result is that, in the SS sequence `1111HHHH2222`,  "1" and "2" are translated to "C". The secondary structure class (`IO.Chain.ssclass`) becomes then `CCCHHHHCCC`; the helix is identified as short when building the secondary structure types (`IO.Chain.sstypes`) so the types sequence becomes `CCCC3333CCCC` that is used to build the topology.

The whole issue comes from a confusion: the secondary structure that is written in the ITP file is not the one that is expected by the `-ss` option. Instead, the `-ss` option expects a string that looks like a sequence produced by `dssp`.

The most backward compatible solution to this issue would be to write the sequence that `-ss` expect in the ITP file, and to make clear that this is the one that is expected. In addition, the `-ss` option should be better documented.

**Comment by Tsjerk** [comment]
The simplest solution at this point is adding 1, 2, and 3 to the translation table, mapping to H. The fix of helix ends, mapping patterns to 1, 2, or 3 is done later, so it would not affect the result.
